packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -218,106 +230,272 @@ export class ModularChangeFamily
 		revInfos: RevisionInfo[],
 		idState: IdAllocationState,
 	): ModularChangeset {
+		const { fieldChanges, nodeChanges, nodeToParent, nodeAliases, crossFieldKeys } =
+			this.composeAllFields(change1.change, change2.change, revInfos, idState);
+
+		const { allBuilds, allDestroys, allRefreshers } = composeBuildsDestroysAndRefreshers([
+			change1,
+			change2,
+		]);
+
+		return makeModularChangeset(
+			this.pruneFieldMap(fieldChanges, nodeChanges),
+			nodeChanges,
+			nodeToParent,
+			nodeAliases,
+			crossFieldKeys,
+			idState.maxId,
+			revInfos,
+			undefined,
+			allBuilds,
+			allDestroys,
+			allRefreshers,
+		);
+	}
+
+	private composeAllFields(
+		change1: ModularChangeset,
+		change2: ModularChangeset,
+		revInfos: RevisionInfo[],
+		idState: IdAllocationState,
+	): ModularChangesetContent {
+		if (hasConflicts(change1) && hasConflicts(change2)) {
+			return {
+				fieldChanges: new Map(),
+				nodeChanges: new Map(),
+				nodeToParent: new Map(),
+				nodeAliases: new Map(),
+				crossFieldKeys: newBTree(),
+			};
+		} else if (hasConflicts(change1)) {
+			return change2;
+		} else if (hasConflicts(change2)) {
+			return change1;
+		}
+
 		const genId: IdAllocator = idAllocatorFromState(idState);
 		const revisionMetadata: RevisionMetadataSource = revisionMetadataSourceFromInfo(revInfos);
 
-		const crossFieldTable = newComposeTable();
+		const crossFieldTable = newComposeTable(change1, change2);
+
+		const composedNodeChanges: ChangeAtomIdMap<NodeChangeset> = mergeNestedMaps(
+			change1.nodeChanges,
+			change2.nodeChanges,
+		);
+
+		const composedNodeToParent = mergeNestedMaps(change1.nodeToParent, change2.nodeToParent);
+
+		const composedNodeAliases: ChangeAtomIdMap<NodeId> = mergeNestedMaps(
+			change1.nodeAliases,
+			change2.nodeAliases,
+		);
```

Comment:
When it comes to node aliases, this seems questionable because it does not result in a union. If X and Y are in the same union in `change1.nodeAliases`, then they are not guaranteed to still be in the same union in `composedNodeAliases`.

Example:
Change 1: `{ X -> Y }`
Change 1: `{ X -> Z }`
composed: either `{ X -> Y }` or `{ X -> Z }`

Instead, we probably want either `{ X -> Y -> Z }` or `{ X -> Z -> Y }`.
I think we can get that by ensuring that whenever we would add `A -> B`, we add `normalized(A) -> B` instead (we can do better perf-wise, but we don't need to care about that right now).